File: src/main/java/Student.java
// Java是如何定义成员变量和静态变量的
public class Student {
    private String name;
    private Integer age;

    // 如果学校都是一样的,那这就是一个全局对象
    private static String school = "atguigu";

    // 此时的构造方法是不需要填入school这个字段的，因为这个字段属于类本身
    public Student(String name, Integer age) {
        this.name = name;
        this.age = age;
    }

    // 提供一个对外接口
    // 这里school不是通过this去获取的，因为school不是定义在对象里的，而是在类里面
    public void getInfo() {
        System.out.println(this.name + this.age + Student.school);
    }


    public static void main(String[] args) {
        Student liukai = new Student("liukai", 29);
        Student xiaoyan = new Student("xiaoyan", 26);
        liukai.getInfo();
        xiaoyan.getInfo();
    }
}

File: src/main/scala/shang_guigu/ch1/Student.scala
package shang_guigu.ch1

// 伴生类，可以访问伴生对象里面的方法
// 在编译之后，伴生类会变成Student.class
class Student (name: String, age: Int){
  def printInfo(): Unit = {
    // 注意name和age是对象本身的，而school是伴生对象的
    println(name + age + Student.school)
  }
}

// 引入伴生对象,把全局对象放到这里
// 在编译之后，伴生对象会变成Student$.class（有dollar符）
object Student {
  val school: String = "atguigu"

  def main(args: Array[String]): Unit = {
    val liukai = new Student("liukai", 28)
    val xiaoyan = new Student("xiaoyan", 26)
    liukai.printInfo()
    xiaoyan.printInfo()
  }
}

/*Java的思路是用static，把类的属性、方法，和对象的属性、方法分隔开
* Scala的思路是用class和object伴生对象，把类和对象区分开
* */